packages/salesforcedx-vscode-core/src/commands/forceSourceStatus.ts

```diff
@@ -77,13 +113,35 @@ const parameterGatherer = new EmptyParametersGatherer();
 export async function forceSourceStatus(
   this: FlagParameter<SourceStatusFlags>
 ) {
-  const {flag, commandVersion} = this || {};
-  const command = commandVersion === CommandVersion.Legacy ? statusCommandLegacy : statusCommand;
-  const executor = new ForceSourceStatusExecutor(flag, command);
-  const commandlet = new SfdxCommandlet(
-    workspaceChecker,
-    parameterGatherer,
-    executor
-  );
-  await commandlet.run();
+  const { flag, commandVersion } = this || {};
+  const command =
+    commandVersion === CommandVersion.Legacy
+      ? statusCommandLegacy
+      : statusCommand;
+
+  if (commandVersion === CommandVersion.Legacy) {
```

Comment:
How about breaking this up into separate functions?  One for CommandVersion.Legacy, and another function for the else case?

Reply:
Yeah, I like that idea.  I'll work on splitting it up.